packages/flutter/lib/src/painting/image_cache.dart

```diff
@@ -47,20 +49,73 @@ class ImageCache {
     _maximumSize = value;
     if (maximumSize == 0) {
       _cache.clear();
+      _currentSizeBytes = 0;
     } else {
-      while (_cache.length > maximumSize)
-        _cache.remove(_cache.keys.first);
+      _checkCacheSize();
     }
   }
 
+  /// The current number of cached entries.
+  int get currentSize => _cache.length;
+
+  /// Maximum size of entries to store in the cache in bytes.
+  ///
+  /// Once more than this amount of bytes have been cached, the
+  /// least-recently-used entry is evicted until there are fewer than the
+  /// maximum bytes.
+  int get maximumSizeBytes => _maximumSizeBytes;
+  int _maximumSizeBytes = _kDefaultSizeBytes;
+  /// Changes the maximum cache bytes.
+  ///
+  /// If the new size is smaller than the current size in bytes, the
+  /// extraneous elements are evicted immediately. Setting this to zero and then
+  /// returning it to its original value will therefore immediately clear the
+  /// cache.
+  set maximumSizeBytes(int value) {
+    assert(value != null);
+    assert(value >= 0);
+    if (value == _maximumSizeBytes)
+      return;
+    _maximumSizeBytes = value;
+    if (_maximumSizeBytes == 0) {
+      _cache.clear();
+      _currentSizeBytes = 0;
+    } else {
+      _checkCacheSize();
+    }
+  }
+
+  /// The current size of cached entries in bytes.
+  int get currentSizeBytes => _currentSizeBytes;
+  int _currentSizeBytes = 0;
+
   /// Evicts all entries from the cache.
   ///
   /// This is useful if, for instance, the root asset bundle has been updated
   /// and therefore new images must be obtained.
-  // TODO(ianh): Provide a way to target individual images. This is currently non-trivial
-  // because by the time we get to the imageCache, the keys we're using are opaque.
   void clear() {
     _cache.clear();
+    _currentSizeBytes = 0;
+  }
+
+  /// Evicts a single entry from the cache.
+  ///
+  /// The [key] must be equal to an object used to cache an image in
+  /// [ImageCache.putIfAbsent].
+  ///
+  /// Prefer using [ImageProvider.evict] to evict images from the cache instead
+  /// of calling method directly.
```

Comment:
we should say why, something like `If the key is not immediately available, as is common, consider using [ImageProvider.evict] to call this method indirectly instead.`